### PR TITLE
Add Callout components and FillLevelContext

### PR DESCRIPTION
**Pull request by dogmar**
[Linear ticket](https://linear.app/pluralsh/issue/ENG-682/implement-callout-component)
[Figma reference](https://www.figma.com/file/9nkKaihJkJzFIgoBADk0yQ/Docs?node-id=8%3A7315)

- Added `FillLevelContext` so that children components can dynamically determine what background lightness they're being rendered on. This is needed for the docs site, since a lot of content is being rendered dynamically, and won't know what `hue` to set manually.
- Added a new `Callout` component that sets its style based on the `FillLevelContext` and is able to set it for its own children.
- Modified `Card` component to both set and be aware of the `FillLevelContext`. If `hue` is not specified, it chooses its styles based on the current `FillLevelContext`.
- Added new `WarningIcon` and updated `ErrorIcon` and `InfoIcon`. These are part of [this ticket](https://linear.app/pluralsh/issue/ENG-711/add-new-icons-to-design-system), but were needed for the `Callout` component.

### CODE REVIEW

**Review by michaeljguarino — APPROVED** [review]

File: src/components/contexts/FillLevelContext.tsx [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+export const FillLevelContext = createContext<FillLevel>(0)
+export const FillLevelProvider = FillLevelContext.Provider
+export const useFillLevel = () => useContext(FillLevelContext)
+
+export const increaseFillLevel: Record<FillLevel, FillLevel> = {
+  0: 1,
+  1: 2,
+  2: 3,
+  3: 3,
+}
```
Comment (maciaszczykm) [inline src/components/contexts/FillLevelContext.tsx:14]: You can also use `Math.min(x + 1, 3)`.

File: src/components/contexts/FillLevelContext.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+export const decreaseFillLevel: Record<FillLevel, FillLevel> = {
+  0: 0,
+  1: 0,
+  2: 1,
+  3: 2,
+}
```
Comment (maciaszczykm) [inline src/components/contexts/FillLevelContext.tsx:21]: You can also use `Math.max(x - 1, 0)`.

File: src/components/Callout.tsx [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+  title,
+  severity = 'info',
+  size = 'full',
+  fillLevel,
+  className,
+  buttonProps,
+  children,
+},
+ref) => {
+  console.log('style', severity)
```
Comment (maciaszczykm) [inline src/components/Callout.tsx:78]: Can you remove it?

**Review by maciaszczykm — COMMENTED** [review]
Looks really cool. I have left some nitpicks.

File: src/components/contexts/FillLevelContext.tsx (hunk 1 above)
Comment (dogmar) [reply src/components/contexts/FillLevelContext.tsx:14]: I was trying to battle the TS checker 🤦‍♂️ with that, but I came up with a better solution now. :) @maciaszczykm PTAL

File: src/components/contexts/FillLevelContext.tsx (hunk 2 above)
Comment (dogmar) [reply src/components/contexts/FillLevelContext.tsx:18]: See above.

**Pull request merged**